src/native/managed/cdac/Microsoft.Diagnostics.DataContractReader.Contracts/Contracts/StackWalk/Context/X86/GCInfoDecoding/CallPattern.cs

```diff
@@ -0,0 +1,115 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.Diagnostics.DataContractReader.Contracts.StackWalkHelpers.X86;
+
+internal static class CallPattern
+{
+
+    /// <summary>
+    /// based on <a href="https://github.com/dotnet/runtime/blob/main/src/coreclr/inc/gcdecoder.cpp">src\inc\gcdecoder.cpp</a> decodeCallPattern
+    /// </summary>
+    public static void DecodeCallPattern(uint pattern, out uint argCnt, out uint regMask, out uint argMask, out uint codeDelta)
+    {
+        uint val = callPatternTable[pattern];
+        byte[] fld = BitConverter.GetBytes(val);
+        argCnt = fld[0];
+        regMask = fld[1];      // EBP,EBX,ESI,EDI
+        argMask = fld[2];
+        codeDelta = fld[3];
+    }
+
+    /// <summary>
+    /// based on <a href="https://github.com/dotnet/runtime/blob/main/src/coreclr/inc/gcdecoder.cpp">src\inc\gcdecoder.cpp</a> callCommonDelta
+    /// </summary>
+    public static readonly uint[] callCommonDelta = [6, 8, 10, 12];
```

Comment:
```suggestion
    public static ReadOnlySpan<uint> callCommonDelta => [6, 8, 10, 12];
```
Can these be readonly spans?